### PR TITLE
ipq806x: add support for NEC Platforms Aterm WG2600HP3

### PR DESCRIPTION
**Pull request by zpc0**
NEC Platforms Aterm WG2600HP3 is a dual-band router based on Qualcomm IPQ8062.

Specification
- SoC: Qualcomm IPQ8062
- RAM: 512MiB
- Flash memory: SPI-NOR 32MiB (Cypress S25FL256S)
- Wi-Fi: Qualcomm QCA9984 (2.4GHz, 1ch - 13ch)
- Wi-Fi: Qualcomm QCA9984 (5GHz, 36ch - 64ch, 100ch - 140ch)
- Ethernet: 4x 100/1000 Mbps (1x WAN, 4x LAN)
- LED: 6x green LED, 6x red LED
- Input: 2x tactile switch, 1x SP3T slide switch
- Serial console: 115200bps, through-hole J3
  - `[  ] [GND] [  ] [TX] [RX] ---->  DC jack`
- Power: DC 12V 1.5A

This device does not support VHT160 and VHT80+80.
Custom BDFs are required to limit VHT capabilities.
https://lists.infradead.org/pipermail/ath10k/2020-July/011551.html

About IPQ8062
--------------------
IPQ8062 is a lower clock variant of IPQ8064.

CPU and NSS clocks:
- CPU: 384 MHz - 1 GHz
- NSS: 110 MHz - 550 MHz

opp and l2 clock values are taken from WG2600HP3 GPL source code [1].

Due to a lack of devices, I didn't test the following features.
- SATA
- NAND flash memory controller
- SD
- USB
- GSBI2, GSBI7
- PCIE2
- GMAC0, GMAC3

Works properly:
- GSBI4 UART
- GSBI5 SPI
- GMAC1, GMAC2
- PCIE0, PCIE1
- MDIO0

Does not work properly:
- CPU SPC
  - This can cause a system hang. Same as IPQ8065.
    See 2336c2dbb1929837f7e42d4315c8073342a5b46b

[1] https://www.aterm.jp/function/wg2600hp3/appendix/opensource.html

There is limited information about this SoC. Any help would be appreciated.

### CODE REVIEW

File: target/linux/ipq806x/image/Makefile [hunk 1]
```diff
@@ -156,6 +156,21 @@ define Device/nec_wg2600hp
 endef
 TARGET_DEVICES += nec_wg2600hp
 
+define Device/nec_wg2600hp3
+	$(call Device/LegacyImage)
+	DEVICE_VENDOR := NEC Platforms
+	DEVICE_MODEL := Aterm WG2600HP3
+	SOC := qcom-ipq8062
+	BLOCKSIZE := 64k
+	BOARD_NAME := wg2600hp3
```
Comment (adschm) [inline target/linux/ipq806x/image/Makefile:165]: I think BOARD_NAME can be dropped.

Comment (zpc0) [reply target/linux/ipq806x/image/Makefile]: Thanks. fixed.

**Comment by zpc0** [comment]
Update:
- remove unneeded BOARD_NAME
- improve commit message

**Comment by zpc0** [comment]
@blocktrron Could you merge [this](https://git.openwrt.org/b9ba6822b5966bb8) into master branch?
I have a similar patch (3be4aac8af3a), but I think your solution is better.

**Comment by zpc0** [comment]
Update:
- Fix incorrect label-mac-device and pci unit-address.

**Comment by zpc0** [comment]
Update:
- Remove unneeded ipq-wifi patch.
- Use linux,default-trigger for Wi-Fi leds.

**Comment by zpc0** [comment]
WG2600HP3 with this patch has been working fine for several weeks, so I will remove WIP status.

File: target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062.dtsi [hunk 2]
```diff
@@ -0,0 +1,101 @@
+#include "qcom-ipq8064.dtsi"
```
Comment (adschm) [inline target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062.dtsi:1]: Please add an SPDX license identifier.

File: target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062-wg2600hp3.dts [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		pinctrl-0 = <&leds_pins>;
+		pinctrl-names = "default";
+
+		led_power_green: power_green {
+			label = "wg2600hp3:green:power";
```
Comment (adschm) [inline target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062-wg2600hp3.dts:66]: I'm just about to get rid of the model: prefix for LED labels, I'd be happy if you drop them here already and get "green:power" etc.

Comment (zpc0) [reply target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062-wg2600hp3.dts]: Thank you for pointing out the recent changes related to label. I dropped model prefix.

File: target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062.dtsi (hunk 2 above)
Comment (zpc0) [reply target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062.dtsi:2]: Added.

**Comment by zpc0** [comment]
Update:
- Drop led model prefix.
- Add SPDX-License-Identifier to qcom-ipq8062.dtsi.

File: target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062.dtsi (hunk 2 above)
Comment (adschm) [reply target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062.dtsi:2]: but in the wrong commit.

Comment (zpc0) [reply target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8062.dtsi]: Sorry for my mistake. It should be fixed now.

**Comment by ynezz** [comment]
Please rebase.

**Comment by zpc0** [comment]
Sure. Rebased.

**Comment by ynezz** [comment]
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

**Comment by zpc0** [comment]
Thank you.

**Pull request closed without merging**